x-pack/plugins/apm/server/lib/services/get_service_transaction_groups_metrics/index.ts

```diff
@@ -0,0 +1,110 @@
+import { getLatencyValue } from '../../helpers/latency_aggregation_type';
+import { Setup, SetupTimeRange } from '../../helpers/setup_request';
+import { getTimeseriesDataForTransactionGroups } from '../get_service_transaction_groups/get_timeseries_data_for_transaction_groups';
+
+export async function getServiceTransactionGroupsMetrics({
+  serviceName,
+  transactionNames,
+  setup,
+  numBuckets,
+  searchAggregatedTransactions,
+  transactionType,
+  latencyAggregationType,
+}: {
+  serviceName: string;
+  transactionNames: string[];
+  setup: Setup & SetupTimeRange;
+  numBuckets: number;
+  searchAggregatedTransactions: boolean;
+  transactionType: string;
+  latencyAggregationType: LatencyAggregationType;
+}): Promise<
+  Record<
+    string,
+    {
+      latency: { timeseries: Coordinate[] };
+      throughput: { timeseries: Coordinate[] };
+      errorRate: { timeseries: Coordinate[] };
+    }
+  >
+> {
+  const { apmEventClient, start, end, esFilter } = setup;
+
+  const buckets = await getTimeseriesDataForTransactionGroups({
+    apmEventClient,
+    start,
+    end,
+    esFilter,
+    numBuckets,
+    searchAggregatedTransactions,
+    serviceName,
+    transactionNames,
+    transactionType,
+    latencyAggregationType,
+  });
+  const deltaAsMinutes = (end - start) / 1000 / 60;
+
+  return buckets.reduce((bucketAcc, bucket) => {
+    const transactionName = bucket.key;
+    return {
+      ...bucketAcc,
+      [transactionName]: bucket.timeseries.buckets.reduce(
+        (acc, timeseriesBucket) => {
+          const x = timeseriesBucket.key;
+          return {
+            ...acc,
+            latency: {
+              timeseries: [
+                ...acc.latency.timeseries,
+                {
+                  x,
+                  y: getLatencyValue({
+                    latencyAggregationType,
+                    aggregation: timeseriesBucket.latency,
+                  }),
+                },
+              ],
+            },
+            throughput: {
+              timeseries: [
+                ...acc.throughput.timeseries,
+                {
+                  x,
+                  y: timeseriesBucket.transaction_count.value / deltaAsMinutes,
+                },
+              ],
+            },
+            errorRate: {
+              timeseries: [
+                ...acc.errorRate.timeseries,
+                {
+                  x,
+                  y:
+                    timeseriesBucket.transaction_count.value > 0
+                      ? (timeseriesBucket[EVENT_OUTCOME].transaction_count
+                          .value ?? 0) /
+                        timeseriesBucket.transaction_count.value
+                      : null,
+                },
+              ],
+            },
+          };
+        },
+        {
+          latency: { timeseries: [] as Coordinate[] },
+          throughput: { timeseries: [] as Coordinate[] },
+          errorRate: { timeseries: [] as Coordinate[] },
```

Comment:
where is impact?

Reply:
@dgieselaar I added the impact here https://github.com/elastic/kibana/pull/88946/commits/aab9c339d68a243cc3135a2da7774ce79b7139fb#diff-df0f4619543ab7f5ec06534330cd13207ecda91a3cfdaab664c2bcd977a2da9bR261, let me know what you think, I had to run another query to calculate the impact for each transaction group, removing the filter on transaction names.